src/optimizer/join_order/cardinality_estimator.cpp

```diff
@@ -107,6 +110,11 @@ void CardinalityEstimator::InitEquivalentRelations(const vector<unique_ptr<Filte
 			continue;
 		} else if (EmptyFilter(*filter)) {
 			continue;
+		} else if ((filter->join_type == JoinType::SEMI || filter->join_type == JoinType::ANTI) && !filter->left_set) {
+			// the right side is from some dummy scan and the comparison is
```

Comment:
this condition can be removed as left_set and right_set are populated when relations are extracted from a semi join